File: 2020-ECP/workflow-examples/example2/io-forwarding.py
#!/usr/bin/env python3

import argparse
import time

parser = argparse.ArgumentParser(description="forward I/O requests for seconds")
parser.add_argument(
    "integer",
    metavar="S",
    type=int,
    help="an integer for the number of seconds to compute",
)

args = parser.parse_args()

print("Will forward I/O requests for " + str(args.integer) + " seconds.")
time.sleep(args.integer)

File: 2026/HPCIC-AWS/tutorial/assets/flux-workflow-examples/job-ensemble/io-forwarding.lua
#!/usr/bin/env lua

local amount = tonumber (arg[1]) or 120

local function sleep (n)
    os.execute ("sleep " .. n)
end

if #arg ~= 1 then
   print ("Usage: io-forward.lua seconds")
   print ("    Forward I/O requests for seconds")
   os.exit (1)
end

print ("Will forward IO requests for " .. amount .. " seconds")
sleep (amount) 

-- vi: ts=4 sw=4 expandtab
